extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -24,7 +24,7 @@ export let KeyedShoppingConfigDef;
  * @return {!Promise<!KeyedShoppingConfigDef>}
  */
 export function getShoppingConfig(pageElement) {
-  const element = pageElement.querySelector('amp-story-shopping-config');
+  const element = pageElement.querySelector('amp-story-shopping-attachment');
```

Comment:
We're creating an unnecessary reference loop by taking a `pageElement` and
querying the `<amp-story-shopping-attachment>` element that makes the call.

I would suggest taking the config element directly instead of querying the page:

```js
/**
  * Gets Shopping config from an element that contains it, by either providing
  * a `src` attribute, or a child matching `<script type="application/json">`
  * The config is validated and keyed by 'product-tag-id'.
  * @param {!Element} element
  * @return {!Promise<!KeyedShoppingConfigDef>}
  */
export function getShoppingConfig(element) {
  return getElementConfig(element).then((config) => {
    //TODO(#36412): Add call to validate config here.
    return keyByProductTagId(config);
  });
}
```

And make an equivalent change in `amp-story-shopping-attachment.js`:

```diff
-   const pageElement = this.element.parentElement;
-   getShoppingConfig(pageElement).then((config) =>
-     storeShoppingConfig(pageElement, config)
+   getShoppingConfig(this.element).then((config) =>
+     storeShoppingConfig(this.element, config)
-   );
```